server/client.go

```diff
@@ -728,29 +750,16 @@ func (c *client) readLoop() {
 			budget = time.Millisecond
 		}
 
-		// Check pending clients for flush.
-		for cp := range c.pcd {
-			// Queue up a flush for those in the set
-			cp.mu.Lock()
-			// Update last activity for message delivery
-			cp.last = last
-			cp.out.fsp--
-			if budget > 0 && cp.flushOutbound() {
-				budget -= cp.out.lft
-			} else {
-				cp.flushSignal()
-			}
-			cp.mu.Unlock()
-			delete(c.pcd, cp)
-		}
+		// Flush, or signal to writeLoop to flush to socket.
+		c.flushClients(budget)
 
 		// Update activity, check read buffer size.
 		c.mu.Lock()
 		nc := c.nc
 
 		// Activity based on interest changes or data/msgs.
 		if c.in.msgs > 0 || c.in.subs > 0 {
-			c.last = last
+			c.last = time.Now()
```

Comment:
You could have flushClients() return the time.Now() that it captures at beginning of the function. It would then be equivalent as previous code in that we would call time.Now() only once.

Reply:
True but I think this is ok.